core/services/relay/evm/chain_reader.go

```diff
@@ -351,20 +343,9 @@ func (cr *chainReader) initDWQuerying(contractName, eventName string, eventDWs m
 }
 
 // getEventItemTypeAndModifier returns codec entry for expected incoming event item and the modifier.
-func (cr *chainReader) getEventItemTypeAndModifier(itemType string, inputMod commoncodec.ModifiersConfig) (types.CodecEntry, commoncodec.Modifier, error) {
+func (cr *chainReader) getEventItemTypeAndModifier(itemType string) (types.CodecEntry, commoncodec.Modifier) {
 	inputTypeInfo := cr.parsed.EncoderDefs[itemType]
-	// TODO can this be simplified? Isn't this same as inputType.Modifier()? BCI-3909
-	inMod, err := inputMod.ToModifier(codec.DecoderHooks...)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	// initialize the modification
-	if _, err = inMod.RetypeToOffChain(reflect.PointerTo(inputTypeInfo.CheckedType()), ""); err != nil {
```

Comment:
This was the only useful part which initialised the internal codec cache for modified types, but I don't think its worth the cognitive complexity since the cache will get populated during the first use anyway